src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -163,7 +166,40 @@ public async Task<IActionResult> GetInsights(string subscriptionId, string resou
             var diagnosticSitePostBody = JsonConvert.DeserializeObject<DiagnosticSiteData>(JsonConvert.SerializeObject(postBody));
             if (IsPostBodyMissing(diagnosticSitePostBody))
             {
-                diagnosticSitePostBody = await GetSitePostBody(subscriptionId, resourceGroupName, siteName);
+                try
+                {
+                    diagnosticSitePostBody = await GetSitePostBody(subscriptionId, resourceGroupName, siteName);
+                }
+                catch (Exception e)
+                {
+                    string path = Request.Path.Value;
+
+                    // For some requests which are querying on the sites that no longer exist. For these cases, instead of
+                    // returnning a 500, we log it to kusto and return a 404.
+                    if (e.Message.StartsWith("Could not get admin sites") || e.Message.StartsWith("Admin Sites response did not contain"))
+                    {
+                        string requestId = null;
+                        if (Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues values) && values != default(StringValues) && values.Count > 0)
+                        {
+                            requestId = values.FirstOrDefault().Split(new char[] { ',' })[0] ?? string.Empty;
+                        }
+                        // either observer adminsites api returns a 404 or the sites api returns do not match the subscriptionId and/or resourceGroupName
+                        Logger.DiagnosticsETWProvider.Instance.LogRuntimeHostHandledException(
+                            requestId,
+                            $"{nameof(SitesController)}.{nameof(GetInsights)}",
+                            subscriptionId,
+                            resourceGroupName,
+                            siteName,
+                            "SiteNotFoundForGetInsightsRequestFromASC",
+                            $"Observer adminsites API could not find the site subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/sites/{siteName}/");
+
+                        return NotFound();
```

Comment:
Instead of returning a 404, can we return a 200 and add a default insight to the response body indicating that the site was not found? A 404 will still impact our SLA, whereas a 200 will not only save the SLA but will also tell the CSS engineer what the problem was and the next steps (Confirm with the customer if the site still exists).

Reply:
Changed 404 to a SiteNotFound insight
![image](https://user-images.githubusercontent.com/68874830/94316564-6de27d80-ff39-11ea-9015-6d07a040a5d0.png)